lib/CodeGen/CGExprScalar.cpp

```diff
@@ -1503,6 +1503,27 @@ Value *ScalarExprEmitter::VisitCastExpr(CastExpr *CE) {
   case CK_LValueToRValue:
     assert(CGF.getContext().hasSameUnqualifiedType(E->getType(), DestTy));
     assert(E->isGLValue() && "lvalue-to-rvalue applied to r-value!");
+
+    if (CE->hasInferredBoundsExpr()) {
+      BoundsExpr* InferredBounds = CE->getInferredBoundsExpr();
+
+      if (InferredBounds->isInvalid())
+        break;
+
+      if (isa<ArraySubscriptExpr>(E)) {
```

Comment:
The subexpression could be a parenthesized expression.   You'll want to ignore parentheses.